alibi/explainers/partial_dependence.py

```diff
@@ -0,0 +1,1294 @@
+                regression: ``'predict'``. Available values for
+                classification: ``'predict_proba'`` | ``'decision_function'``. The choice should be considered
+                in analogy with the `sklearn` estimators API and the `response_method` used in
+                :py:meth:`alibi.explainers.partial_dependence.explain`.
+
+             - ``'num_classes'`` : ``Optional[int]`` - Number of classes predicted by the `predictor` function.
+                Considered only for ``prediction_type='classification'``.
+
+        """
+        super().__init__(meta=copy.deepcopy(DEFAULT_META_PD))
+        self.feature_names = feature_names
+        self.categorical_names = categorical_names
+        self.target_names = target_names
+
+        if isinstance(predictor, BaseEstimator):
+            self.predictor = predictor
+        elif callable(predictor):
+            if predictor_kw is None:
+                predictor_kw = {}
+            self.predictor = PDEstimatorWrapper(predictor=predictor, **predictor_kw)
+        else:
+            raise ValueError(f'Unknown predictor type. Received {type(predictor)}. Supported type are: '
+                             f'sklearn.base.BaseEstimator | Callable[[np.ndarray], np.ndarray].')
+
+    def explain(self,  # type: ignore[override]
+                X: np.ndarray,
+                features_list: Optional[List[Union[int, Tuple[int, int]]]] = None,
+                response_method: Literal['auto', 'predict_proba', 'decision_function'] = 'auto',
+                method: Literal['auto', 'recursion', 'brute'] = 'auto',
+                kind: Literal['average', 'individual', 'both'] = 'average',
+                percentiles: Tuple[float, float] = (0., 1.),
+                grid_resolution: int = 100,
+                grid_points: Optional[Dict[int, np.ndarray]] = None) -> Explanation:
+        """
+        Calculates the partial dependence for each feature and/or pairs of features with respect to the all targets
+        and the reference dataset `X`.
+
+        Parameters
+        ----------
+        X
+            An `N x F` reference tabular dataset used to calculate partial dependence curves. This is typically the
+            training dataset or a representative sample.
+        features_list
+            An optional list of features or pairs of features for which to calculate the partial dependence for.
+            If not provided, the partial dependence will be computed for every single features in the dataset.
+        response_method
+            Specifies the prediction function to be used. For a classifier it specifies whether to use the
+            `predict_proba` or the `decision_function`. For a regressor, the parameter is ignored. If set to `auto`,
+            the `predict_proba` is tried first, and if not supported then it reverts to `decision_function`. Note
+            that if `method='recursion'`, the prediction function always uses `decision_function`.
```

Comment:
It doesn't relate only to `sklearn` estimators. Once a blackbox model is wrapped, there is no longer a difference between the `sklearn` models and the bb.

Reply:
This is true, but I think we need more documentation about this, perhaps at the level of `predictor_kw` in the constructor and then cross-referencing between the `explain` and the constructor, as it's not fully clear in the black-box case.

This is similar to the previous discussion of having to specify `predictor_kw` like `prediction_fn` for black-box cases where it can be confusing what people are asked to do.